cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -640,12 +639,24 @@ public void destroy() {
                               .collect(Collectors.toSet()));
       }
 
+      List<FieldOperation> fieldOperations = allStageOperations.get(stageSpec.getName());
+      // if the field operations are empty for a stage, auto generate it for the stages that have input schema and
+      // output schema
+      if (fieldOperations.isEmpty()) {
+        // simply add to the list since this list is initialized in previous lines
+        fieldOperations.add(
+          // the name will be appended by the stage name in later operations
+          new FieldTransformOperation("Transform",
+                                      String.format("Generated operation for stage %s", stageSpec.getName()),
```

Comment:
'Generated operation' seems like it would be hard to understand. I'm also not sure it's a great idea to include the stage name, since that is an aribitrary value chosen by the user, which doesn't have to describe what the stage is doing. It may be fine to just leave this as an empty string.